### DISCUSSION TITLE
ACL Implemetation on Topic Level

### DISCUSSION THREAD

**Discussion by sejalgoy** [strimzi/strimzi-kafka-operator#7606] [Q&A]
Hi Everyone,

I have a Kafka cluster on strimzi. I want to implement ACL/Authorization on Topic level. I have read about Simple, Keycloak and OPA..
Not able to figure out if Keycloak or OPA which one is better.
Could you please help me in this?

**Comment by scholzj** [comment]
WDYM with ACLs on a _topic level_?

**Reply by sejalgoy** [reply]
May i know what is WDYM?

**Reply by scholzj** [reply]
Ahh, sorry ... WDYM means _what do you mean_ ;-).

All the authorization types allow you to define different rules for different topics. So I wonder if you have some specific features you need or what exactly do you mean with _Topic Level authorization_.

Apart from that ... it is about what do you use. Keycloak authorization requires you to use Keycloak and OAuth authentication. OPA can do anything. But you actually need to write the policies.

**Reply by sejalgoy** [reply]
In our organization, we are not allowed to use Keycloak server, can we implement Authentication and OAUTH authorization with Azure Authorization?

**Reply by scholzj** [reply]
AFAIK, OAuth itself does not offer any proper solution for authorization which would be usable with Kafka. The Keycloak authorization support is using some Keycloak specific features (Keycloak Authorization Services). If Azure has something similar, it is likely not compatible with the Keycloak feature. So no, I don't think you can use that.

**Reply by sejalgoy** [reply]
Thank you!
